### PR TITLE
Avocado logging

### PR DESCRIPTION
**Pull request by psklenar**
https://github.com/fedora-modularity/meta-test-family/issues/85

it adds summary to mtf command
better logging
tool avocado_log_json.py can read json genereted by avocado

### CODE REVIEW

**Comment by psklenar** [comment]
see new log:
```
2017-09-19 15:42:19,143 extension        L0157 DEBUG| found extension EntryPoint.parse('journal = avocado.plugins.journal:JournalResult')
2017-09-19 15:42:19,143 extension        L0157 DEBUG| found extension EntryPoint.parse('tap = avocado.plugins.tap:TAPResult')
2017-09-19 15:42:19,144 extension        L0157 DEBUG| found extension EntryPoint.parse('human = avocado.plugins.human:Human')
2017-09-19 15:42:19,144 extension        L0157 DEBUG| found extension EntryPoint.parse('resultsdb = avocado_resultsdb:ResultsdbResult')
2017-09-19 15:42:19,145 extension        L0157 DEBUG| found extension EntryPoint.parse('teststmpdir = avocado.plugins.teststmpdir:TestsTmpDir')
2017-09-19 15:42:19,145 extension        L0157 DEBUG| found extension EntryPoint.parse('jobscripts = avocado.plugins.jobscripts:JobScripts')
2017-09-19 15:42:19,145 extension        L0157 DEBUG| found extension EntryPoint.parse('human = avocado.plugins.human:HumanJob')
2017-09-19 15:42:19,149 extension        L0157 DEBUG| found extension EntryPoint.parse('yaml_to_mux = avocado_varianter_yaml_to_mux:YamlToMux')
2017-09-19 15:42:19,149 sysinfo          L0413 INFO | Commands configured by file: /etc/avocado/sysinfo/commands
2017-09-19 15:42:19,150 sysinfo          L0424 INFO | Files configured by file: /etc/avocado/sysinfo/files
2017-09-19 15:42:19,150 sysinfo          L0444 INFO | Profilers configured by file: /etc/avocado/sysinfo/profilers
2017-09-19 15:42:19,150 sysinfo          L0452 INFO | Profiler disabled
2017-09-19 15:42:19,150 sysinfo          L0502 INFO | System log file not found (looked for ['/var/log/messages', '/var/log/syslog', '/var/log/system.log'])
2017-09-19 15:42:19,158 job              L0321 INFO | Command line: /bin/avocado run copyTest.py escapingTest.py microdnfinside.py ../../moduleframework/tools/check_compose.py ../../moduleframework/tools/modulelint.py ../../moduleframework/tools/rpmvalidation.py PureAvocadoTest.py rpmvalidator.py shellTest.sh simpleTest.py skipTest.py
2017-09-19 15:42:19,158 job              L0322 INFO | 
2017-09-19 15:42:19,158 job              L0326 INFO | Avocado version: 52.0
2017-09-19 15:42:19,158 job              L0342 INFO | 
2017-09-19 15:42:19,158 job              L0346 INFO | Config files read (in order):
2017-09-19 15:42:19,158 job              L0348 INFO | /etc/avocado/avocado.conf
2017-09-19 15:42:19,158 job              L0348 INFO | /etc/avocado/conf.d/gdb.conf
2017-09-19 15:42:19,158 job              L0348 INFO | /root/.config/avocado/avocado.conf
2017-09-19 15:42:19,158 job              L0353 INFO | 
2017-09-19 15:42:19,158 job              L0355 INFO | Avocado config:
2017-09-19 15:42:19,159 job              L0364 INFO | Section.Key                             Value
2017-09-19 15:42:19,159 job              L0364 INFO | datadir.paths.base_dir                  /var/lib/avocado
2017-09-19 15:42:19,159 job              L0364 INFO | datadir.paths.test_dir                  /usr/share/avocado/tests
2017-09-19 15:42:19,159 job              L0364 INFO | datadir.paths.data_dir                  /var/lib/avocado/data
2017-09-19 15:42:19,159 job              L0364 INFO | datadir.paths.logs_dir                  ~/avocado/job-results
2017-09-19 15:42:19,159 job              L0364 INFO | sysinfo.collect.enabled                 True
2017-09-19 15:42:19,159 job              L0364 INFO | sysinfo.collect.commands_timeout        -1
2017-09-19 15:42:19,159 job              L0364 INFO | sysinfo.collect.installed_packages      False
2017-09-19 15:42:19,159 job              L0364 INFO | sysinfo.collect.profiler                False
2017-09-19 15:42:19,159 job              L0364 INFO | sysinfo.collect.locale                  C
2017-09-19 15:42:19,159 job              L0364 INFO | sysinfo.collectibles.commands           /etc/avocado/sysinfo/commands
2017-09-19 15:42:19,159 job              L0364 INFO | sysinfo.collectibles.files              /etc/avocado/sysinfo/files
2017-09-19 15:42:19,159 job              L0364 INFO | sysinfo.collectibles.profilers          /etc/avocado/sysinfo/profilers
2017-09-19 15:42:19,159 job              L0364 INFO | runner.output.colored                   True
2017-09-19 15:42:19,159 job              L0364 INFO | runner.output.utf8                      
2017-09-19 15:42:19,159 job              L0364 INFO | remoter.behavior.reject_unknown_hosts   False
2017-09-19 15:42:19,159 job              L0364 INFO | remoter.behavior.disable_known_hosts    False
2017-09-19 15:42:19,159 job              L0364 INFO | job.output.loglevel                     debug
2017-09-19 15:42:19,159 job              L0364 INFO | restclient.connection.hostname          localhost
2017-09-19 15:42:19,159 job              L0364 INFO | restclient.connection.port              9405
2017-09-19 15:42:19,159 job              L0364 INFO | restclient.connection.username          
2017-09-19 15:42:19,159 job              L0364 INFO | restclient.connection.password          
2017-09-19 15:42:19,159 job              L0364 INFO | plugins.disable                         []
2017-09-19 15:42:19,160 job              L0364 INFO | plugins.skip_broken_plugin_notification []
2017-09-19 15:42:19,160 job              L0364 INFO | plugins.loaders                         ['file', '@DEFAULT']
2017-09-19 15:42:19,160 job              L0364 INFO | gdb.paths.gdb                           /usr/bin/gdb
2017-09-19 15:42:19,160 job              L0364 INFO | gdb.paths.gdbserver                     /usr/bin/gdbserver
2017-09-19 15:42:19,160 job              L0365 INFO | 
2017-09-19 15:42:19,160 job              L0368 INFO | Avocado Data Directories:
2017-09-19 15:42:19,160 job              L0369 INFO | 
2017-09-19 15:42:19,160 job              L0370 INFO | base     /var/lib/avocado
2017-09-19 15:42:19,161 job              L0371 INFO | tests    /usr/share/avocado/tests
2017-09-19 15:42:19,161 job              L0372 INFO | data     /var/lib/avocado/data
2017-09-19 15:42:19,161 job              L0373 INFO | logs     /root/avocado/job-results/job-2017-09-19T15.42-febb8fb
2017-09-19 15:42:19,161 job              L0374 INFO | 
2017-09-19 15:42:19,161 job              L0379 INFO | No variants available, using defaults only
2017-09-19 15:42:19,161 job              L0379 INFO | 
2017-09-19 15:42:19,161 job              L0379 INFO | Variant :    /
2017-09-19 15:42:19,161 job              L0382 INFO | Temporary dir: /var/tmp/avocado__WU2Sn
2017-09-19 15:42:19,161 job              L0383 INFO | 
2017-09-19 15:42:19,161 job              L0313 INFO | Job ID: febb8fbe243537c9aa94f7944866f2f17467b0bb
2017-09-19 15:42:19,161 job              L0316 INFO | 
2017-09-19 15:42:19,313 sysinfo          L0109 DEBUG| Not logging /proc/pci (file does not exist)
2017-09-19 15:42:19,383 sysinfo          L0413 INFO | Commands configured by file: /etc/avocado/sysinfo/commands
2017-09-19 15:42:19,383 sysinfo          L0424 INFO | Files configured by file: /etc/avocado/sysinfo/files
2017-09-19 15:42:19,383 sysinfo          L0444 INFO | Profilers configured by file: /etc/avocado/sysinfo/profilers
2017-09-19 15:42:19,383 sysinfo          L0452 INFO | Profiler disabled
2017-09-19 15:42:19,384 sysinfo          L0502 INFO | System log file not found (looked for ['/var/log/messages', '/var/log/syslog', '/var/log/system.log'])
2017-09-19 15:42:19,392 varianter        L0112 DEBUG| PARAMS (key=timeout, path=*, default=None) => None
2017-09-19 15:42:19,392 test             L0240 INFO | START 01-copyTest.py:CheckCopyFiles.testCopyThereAndBack
2017-09-19 15:42:21,294 output           L0655 DEBUG| Module Type: docker; Profile: default
2017-09-19 15:42:21,332 avocado_test     L0080 DEBUG|  
2017-09-19 15:42:21,332 avocado_test     L0081 DEBUG|  
2017-09-19 15:42:21,332 avocado_test     L0082 DEBUG| 

:::::::::::::::::::::::: START ::::::::::::::::::::::::

2017-09-19 15:42:21,332 avocado_test     L0083 DEBUG|  
2017-09-19 15:42:21,332 avocado_test     L0084 DEBUG|  
2017-09-19 15:42:24,249 avocado_test     L0109 DEBUG| 

:::::::::::::::::::::::: START MODULE ::::::::::::::::::::::::

2017-09-19 15:42:24,613 avocado_test     L0184 DEBUG| 

:::::::::::::::::::::::: COMMAND ON HOST <!> ::::::::::::::::::::::::

2017-09-19 15:42:24,613 process          L0389 INFO | Running 'rm a b'
2017-09-19 15:42:24,617 process          L0479 DEBUG| [stderr] rm: cannot remove 'a': No such file or directory
2017-09-19 15:42:24,617 process          L0499 INFO | Command 'rm a b' finished with 1 after 0.000583171844482s
2017-09-19 15:42:24,617 process          L0479 DEBUG| [stderr] rm: cannot remove 'b': No such file or directory
2017-09-19 15:42:24,618 avocado_test     L0133 DEBUG| 

:::::::::::::::::::::::: COMMAND IN MODULE <-> ::::::::::::::::::::::::

2017-09-19 15:42:24,618 process          L0389 INFO | Running 'docker exec 7975e979f0ea63470104909d4c46250df9e88901fbbc73704fae461ed91c8e40 bash -c "rm /a.test"'
2017-09-19 15:42:24,667 process          L0479 DEBUG| [stderr] rm: cannot remove '/a.test': No such file or directory
2017-09-19 15:42:24,669 process          L0499 INFO | Command 'docker exec 7975e979f0ea63470104909d4c46250df9e88901fbbc73704fae461ed91c8e40 bash -c "rm /a.test"' finished with 1 after 0.0477409362793s
2017-09-19 15:42:24,669 avocado_test     L0184 DEBUG| 

:::::::::::::::::::::::: COMMAND ON HOST <!> ::::::::::::::::::::::::

2017-09-19 15:42:24,669 process          L0389 INFO | Running 'echo x > a'
2017-09-19 15:42:24,674 process          L0499 INFO | Command 'echo x > a' finished with 0 after 0.00100803375244s
2017-09-19 15:42:24,903 avocado_test     L0133 DEBUG| 

:::::::::::::::::::::::: COMMAND IN MODULE <-> ::::::::::::::::::::::::

2017-09-19 15:42:24,904 process          L0389 INFO | Running 'docker exec 7975e979f0ea63470104909d4c46250df9e88901fbbc73704fae461ed91c8e40 bash -c "cat /a.test"'
2017-09-19 15:42:24,960 process          L0479 DEBUG| [stdout] x
2017-09-19 15:42:24,962 process          L0499 INFO | Command 'docker exec 7975e979f0ea63470104909d4c46250df9e88901fbbc73704fae461ed91c8e40 bash -c "cat /a.test"' finished with 0 after 0.0538568496704s
2017-09-19 15:42:25,077 avocado_test     L0184 DEBUG| 

:::::::::::::::::::::::: COMMAND ON HOST <!> ::::::::::::::::::::::::

2017-09-19 15:42:25,077 process          L0389 INFO | Running 'cat b'
2017-09-19 15:42:25,082 process          L0479 DEBUG| [stdout] x
2017-09-19 15:42:25,082 process          L0499 INFO | Command 'cat b' finished with 0 after 0.000543832778931s
2017-09-19 15:42:25,082 avocado_test     L0184 DEBUG| 

:::::::::::::::::::::::: COMMAND ON HOST <!> ::::::::::::::::::::::::

2017-09-19 15:42:25,082 process          L0389 INFO | Running 'rm a b'
2017-09-19 15:42:25,086 process          L0499 INFO | Command 'rm a b' finished with 0 after 0.000547170639038s
2017-09-19 15:42:25,087 avocado_test     L0096 DEBUG| 

:::::::::::::::::::::::: TEARDOWN ::::::::::::::::::::::::

2017-09-19 15:42:35,463 test             L0750 INFO | PASS 01-copyTest.py:CheckCopyFiles.testCopyThereAndBack
2017-09-19 15:42:35,463 test             L0733 INFO | 
2017-09-19 15:42:35,522 sysinfo          L0413 INFO | Commands configured by file: /etc/avocado/sysinfo/commands
2017-09-19 15:42:35,522 sysinfo          L0424 INFO | Files configured by file: /etc/avocado/sysinfo/files
2017-09-19 15:42:35,522 sysinfo          L0444 INFO | Profilers configured by file: /etc/avocado/sysinfo/profilers
2017-09-19 15:42:35,522 sysinfo          L0452 INFO | Profiler disabled
2017-09-19 15:42:35,522 sysinfo          L0502 INFO | System log file not found (looked for ['/var/log/messages', '/var/log/syslog', '/var/log/system.log'])
2017-09-19 15:42:35,531 varianter        L0112 DEBUG| PARAMS (key=timeout, path=*, default=None) => None
2017-09-19 15:42:35,531 test             L0240 INFO | START 02-escapingTest.py:SanityCheckApostophes.testSimpleShell
```

see mtf command how it ends now during ERROR
```
TEST:   01-copyTest.py:CheckCopyFiles.testCopyThereAndBack
ERROR:  Command 'docker exec 2e4421c65dfa830254e7d6d54aeee31276366610788acd31c2ab80590942c23e bash -c "cat /a.testXXXXXXXXXXXXXXXx"' failed (rc=1)
LOG:    /root/avocado/job-results/job-2017-09-21T14.02-002eae0/test-results/01-copyTest.py:CheckCopyFiles.testCopyThereAndBack/debug.log
```

File: moduleframework/avocado_testers/avocado_test.py [hunk 1]
```diff
@@ -75,6 +77,11 @@ def setUp(self):
 
         :return: None
         """
+        self.log.debug(" ")
+        self.log.debug(" ")
+        self.log.debug("{0} START {1}".format(self.delimiterStart, self.delimiterEnd))
```
Comment (jscotka) [inline moduleframework/avocado_testers/avocado_test.py]: rename start to SETUP

File: moduleframework/avocado_testers/avocado_test.py [hunk 2]
```diff
@@ -75,6 +77,11 @@ def setUp(self):
...
+        self.log.debug(" ")
```
Comment (jscotka) [inline moduleframework/avocado_testers/avocado_test.py]: remove this line

File: moduleframework/avocado_testers/avocado_test.py [hunk 3]
```diff
@@ -75,6 +77,11 @@ def setUp(self):
...
+        self.log.debug(" ")
```
Comment (jscotka) [inline moduleframework/avocado_testers/avocado_test.py]: remove this line

File: moduleframework/avocado_testers/avocado_test.py [hunk 4]
```diff
@@ -60,6 +60,11 @@ def __init__(self, *args, **kwargs):
             "Module Type: %s; Profile: %s" %
             (self.moduleType, self.moduleProfile))
 
+    def __print_breaks(self, message):
+        delimiterStart = '\n\n::::::::::::::::::::::::'
```
Comment (phracek) [inline moduleframework/avocado_testers/avocado_test.py]: Move backslash into `self.log.debug` fce and delimiter can be used only once.

File: tools/avocado_log_json.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        # file is not readable as json: No JSON object could be decoded
+        print(e)
+        exit(1)
+    except:
+        print("no file: specify 1 argument as existing json file")
+        exit(3)
+    delimiter=""
+    for i in data['tests']:
+        if i.get('status') == 'ERROR':
+            print(delimiter)
```
Comment (phracek) [inline tools/avocado_log_json.py]: This can be only print("").

File: tools/avocado_log_json.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            print("TEST:   {0}".format(i.get('id')))
+            print("ERROR:  {0}".format(i.get('fail_reason')))
+            print("LOG:    {0}".format(i.get('logfile')))
+            delimiter = "-------------------------"
```
Comment (phracek) [inline tools/avocado_log_json.py]: Move outside of loop.

File: tools/mtf [hunk 7]
```diff
@@ -24,7 +25,7 @@ while [[ $# -gt 0 ]]; do
         exit 0
       ;;
       -l|--linters)
-        AVOCADO_ARGS="$AVOCADO_ARGS $SITE_LIB/$MTF_TOOLS/*.py"
+        AVOCADO_ARGS="$AVOCADO_ARGS $MTF_TOOLS/modulelint/*.py"
```
Comment (phracek) [inline tools/mtf]: This is wrong. All modulelint are mentioned in site lib. 
We move `/usr/share/moduleframework/tools` to `/usr/lib/python2.7/site-packages/moduleframework/tools`.

File: tools/mtf [hunk 8]
```diff
@@ -34,3 +35,5 @@ while [[ $# -gt 0 ]]; do
 done
 
 avocado run $AVOCADO_ARGS
+$MTF_TOOLS/avocado_log_json.py $JSON_LOG
```
Comment (phracek) [inline tools/mtf]: Move to `site-packages` directory or create a script called `mtf-log-json` or similar one.

File: tools/avocado_log_json.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+#
+
+"""
+Utility for reading avocado json files.
+"""
+
+import sys
+import json
+
+def main():
```
Comment (phracek) [inline tools/avocado_log_json.py]: Main is good for binaries like `/usr/bin/mtf-avocado-log-json`. Move it from `/usr/share/moduleframework/tools` to `site-packages`

File: tools/mtf (hunk 7 above)
Comment (jscotka) [reply tools/mtf]: We've dicsussed it. It is fixed in devel branch and repaired previous code SITE_LIB code what didn't work.
Actually merge has to be done carefully manually. (it is in conflict with devel) or rebased here.

File: tools/avocado_log_json.py (hunk 6 above)
Comment (psklenar) [reply tools/avocado_log_json.py]: its inside loop due to writing each error with its log separately, also delimiter is due to better logging first error with empty line, next with ----

@phracek : Or you mean write only one error : the first one with its log?

File: tools/avocado_log_json.py (hunk 5 above)
Comment (psklenar) [reply tools/avocado_log_json.py]: delimiter is due to better visibility of logs; the first error with empty line, next with ---------

File: tools/mtf (hunk 8 above)
Comment (psklenar) [reply tools/mtf]: done

File: tools/mtf (hunk 7 above)
Comment (psklenar) [reply tools/mtf]: done

File: tools/avocado_log_json.py (hunk 9 above)
Comment (psklenar) [reply tools/avocado_log_json.py]: done

**Pull request merged**